src/app/todo-list/interfaces/vendor.interface.ts

```diff
@@ -0,0 +1,102 @@
+import { ItemType } from '@app/service/model';
+
+/**
+ * Represents the `vendors` endpoint with the `vendorSales` component
+ */
+export interface VendorResponse {
+  categories: Privacy;
+  currencyLookups: Privacy;
+  itemComponents: { [key: string]: any; } // in my experience, any is usually `null`
+  sales: {
+    data: VendorsSalesList // key is the vendor hash
+  }
+  vendorGroups: Privacy;
+  vendors: Privacy;
+}
+
+export interface VendorsSalesList {
+  saleItems: { [key: string]: VendorSales; } // key is an arbitrary number
+}
+
+export interface VendorSales {
+  saleItems: { [key: string]: SaleItem; } // key is an arbitrary number
+}
+
+export interface SaleItem {
+  augments: number;
+  costs: CostReward[];
+  failureIndexes: any[];
+  itemHash: number; // used to look up what the item is
+  quantity: number;
+  saleStatus: SaleStatus;
+  vendorItemIndex: number;
+}
+
+export interface CostReward {
+  itemHash: number;
+  quantity: number;
+}
+
+export interface Reward {
+  itemValue: CostReward[];
+  valueDescription: string;
+}
+
+export interface Vendor {
+  hash: number;
+  displayProperties: DisplayProperties;
+}
+
+export interface DisplayProperties {
+  description: string;
+  icon: string;
+  name: string;
+}
+
+/**
+ * There are many more properties, but these are the most useful to us,
+ * where we only care about bounties at the moment
+ */
+export interface InventoryItem {
+  displayProperties: DisplayProperties;
+  itemTypeDisplayName: string; // "Daily Bounty"
+  itemTypeAndTierDisplayName: string; // "Common Daily Bounty"
+  inventory: any; // could be useful for the inventory item name?
+  value: Reward; // rewards
+  objectives: any; // TODO: track objectives
+  itemCategoryHashes: number[]; // [3441456675, 1784235469] which equates to "Bounties: Daily" and "Bounties" 
+  itemType: ItemType; // <enum "Bounty" 26>
+  traitIds?: string[]; // ["inventory_filtering.bounty", "item_type.bounty"] seems redundant to `itemCategoryHashes`
+  hash: number; // 304697572
+  index: number; // 14618
+}
+
+/**
+ * Idk if this is useful, but it is very common
+ */
+export interface Privacy {
+  privacy: number
+}
+
+/**
+ * This is defined by us, a combination of the manifest bounty,
+ * the character progressions on the bounty, and the vendor
+ */
+export interface Bounty extends InventoryItem {
+  costs: CostReward[]; // assuming that it's the same cost for each character
+  vendorName: string;
+  chars: { [key: string]: BountyCharInfo; }
+}
+
+export interface BountyCharInfo {
+  saleStatus: SaleStatus;
+  progress?: any; // TODO: track progress and define a better interface
+}
+
+export enum SaleStatus {
+  AVAILABLE = 0,
+  COMPLETED = 8,
+  NOT_FOR_SALE = 32,
+  NOT_AVAILABLE = 64,
+  ALREADY_HELD = 128,
+}
```

Comment:
Most of the things in this file are exposing properties already on the objects we get back from the API or the manifest.